raft/raftpb/raft.proto

```diff
@@ -22,6 +22,7 @@ message Entry {
 	optional uint64     Term  = 2 [(gogoproto.nullable) = false]; // must be 64-bit aligned for atomic operations
 	optional uint64     Index = 3 [(gogoproto.nullable) = false]; // must be 64-bit aligned for atomic operations
 	optional EntryType  Type  = 1 [(gogoproto.nullable) = false];
+	optional uint64     ID    = 5 [(gogoproto.nullable) = false];
```

Comment:
I'm afraid that modifying raft protocol have very high bar and particular I think
the main reason we need to do this (as etcd) is to preserve backward compatibility of the 'Data' field. 

I think its good for experimenting / showing impact of the change but 
if we decide to go with the change, we should find other mean to pass the data through raft. 
It might be as ugly as appending a special `datamarker + ID` at the end of the Data (as proto knows its size and usually does not parse beyond the size).

Reply:
yeah, I agree. We may need to find some other approach to do this. We can see there is a potential gain by reducing the unnecessary marshaling and unmarshalling.

Reply:
The tests were performed on 1 instance etcd cluster or 3-node ? 
I wonder whether the performance improvement diminish if we have 1 node faster (proposer), 
but actually the 2 other are slower to commit, so all in all performance does not changes.

Reply:
> The tests were performed on 1 instance etcd cluster or 3-node ?
> I wonder whether the performance improvement diminish if we have 1 node faster (proposer),
> but actually the 2 other are slower to commit, so all in all performance does not changes.

Yes, this is possible. I think I need to do some cluster performance analysis now.